AutoMad/autoMad.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Oct 22 13:14:59 2024
+
+@author: dragon
+"""
+
+import random
+import re
+import wordGrabber
+import madLibFeeder
+import madDictionary
+import nltk
+import pandas as pd
+import numpy as np
+from nltk.corpus import stopwords
+from nltk.tokenize import word_tokenize, sent_tokenize
+
+NOUNS = ["NN","NNS"]
+VERBS = ["EX","MD","VB","VBD","VBG","VBN","VBP","VBZ"]
+CONJUNCTIONS =["CC"]
+NUMBER = ["CD","LS"]
+ADJECTIVES = ["JJ","JJR","JJS","WDT"]
+ADVERBS = ["RB","RBR","RBS","RP","WRB"]
+SILLIES = ["UH"]
+UNKNOWN = ["FW","PDT","POS"]
+PREPOSITIONS = ["IN","TO","WP","WP$"]
+PRONOUNS = ["PRP","PRP$"]
+ARTICLES = ["DT"]
+NAMES = ["NNP","NNPS"]
+
+def posToCategory(pos):
+    if pos in NOUNS:
+        return 'noun'
+    if pos in VERBS:
+        return 'verb'
+    if pos in CONJUNCTIONS:
+        return 'conjunction'
+    if pos in NUMBER:
+        return 'number'
+    if pos in ADJECTIVES:
+        return 'adjective'
+    if pos in ADVERBS:
+        return 'adverb'
+    if pos in SILLIES:
+        return 'sillyword'
+    if pos in UNKNOWN:
+        return 'sillyword'
+    if pos in PREPOSITIONS:
+        return 'preposition'
+    if pos in PRONOUNS:
+        return 'pronoun'
+    if pos in ARTICLES:
+        return 'article'
+    if pos in NAMES:
+        return 'name'
+    return pos + "word Not Found"
+        
+def autoMad(story, percent, dictionary):
```

Comment:
I'd encourage you to add type annotations to all of your functions. It will make it easier to keep track of what is being passed in (i.e. it took me a little while to figure out that the `dictionary` argument is probably an instance of your `madDictionary` class, and not a python `dict`)

Then I'd also encourage you to get in the habit of running [mypy](https://github.com/python/mypy) frequently, it will catch some errors for you (kind of like you would get by compiling a java program)